feedback_prompt_learning/search_algo/world_model.py

```
@@ -0,0 +1,399 @@
+            )
+        else:
+            output = self.reward_fn(cleaned_output, target, prompt)
+
+        # Handle both old tuple format and new RewardOutput
+        if isinstance(output, RewardOutput):
+            return output
+        elif isinstance(output, tuple):
+            # Legacy format: (score, feedback_dict)
+            return RewardOutput.from_tuple(output)
+        else:
+            raise TypeError(f"Reward function must return RewardOutput or (float, dict) tuple, got {type(output)}")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)